### PR TITLE
feat(quickopen): Add binds to open in splits to Quick Open

### LINKED ISSUE #564: Commands: QuickOpen - open in vertical split / open in horizontal split
For the QuickOpen, it is often convenient to be able to open directly in a split.

We should have shortcut keys out-of-the-box for this, as well as have them be bindable commands.

Proposal:
- `quickOpen.verticalSplit` - opens the currently selected file in a vertical split (Default: `<c-v>`)
- `quickOpen.horizontalSplit` - opens the currently selected file in a horizontal split (Default: `<c-s>`)

These commands would only be enabled when the quickopen menu is up

### LINKED ISSUE #2232: QuickOpen: Open in new (layout) tab
When working with layout tabs

```json
{
  "oni.layout.layoutTabPosition": "top",
  "oni.layout.showLayoutTabs": "always",
  "workbench.editor.showTabs": false
}
```

it would be super useful to have a  `<c-t>` key binding in QuickOpen to open the selected file in a new tab. Otherwise, files will get opened in the same tab and opening a file with QuickOpen becomes a cumbersome operation.

### PR DESCRIPTION
**Pull request by CrossR**
Fix #564
Fix #2232

A rough first pass, but adds default bindings for selecting items from the quick open to open in current/vertical/horizontal/new tab.

Binds aren't finalised, so some input there would be nice (Ctrl-s and Ctrl-v are the defaults in Fzf in vim which is why I went with them).

### CODE REVIEW

File: src/Model/GlobalCommands.re [hunk 1]
```diff
@@ -43,7 +43,33 @@ module List = {
...
+        direction: Oni_Core.SplitDirection.Vertical({shouldReuse: false}),
+      }),
+    );
+  let selectNewTab =
+    register(
+      "list.selectNewTab",
+      ListSelect({
+        direction: Oni_Core.SplitDirection.NewTab,
+      }),
+    );
```
Comment (bryphe) [inline src/Model/GlobalCommands.re:71]: Awesome! Thanks for adding these 🎉

File: src/Model/State.re [hunk 2]
```diff
@@ -125,6 +125,21 @@ let defaultKeyBindings =
         ~command=Commands.List.select.id,
         ~condition="listFocus" |> WhenExpr.parse,
       ),
+      bind(
+        ~key="<C-v>",
+        ~command=Commands.List.selectVertical.id,
+        ~condition="listFocus" |> WhenExpr.parse,
+      ),
```
Comment (bryphe) [inline src/Model/State.re:132]: The only concern I have here is that the `Control+V` keybinding would conflict with paste on Windows/Linux (ie, regressing https://github.com/onivim/oni2/issues/2308)

A couple ideas:
- We could use `Control+Shift+V` for opening in vertical tab
- We could use the `Shift+CR` keybinding for vertical, and `Control+T` for new tab

File: src/Model/GlobalCommands.re [hunk 3]
```diff
@@ -43,7 +43,33 @@ module List = {
...
+      "list.selectVertical",
+      ListSelect({
+        direction: Oni_Core.SplitDirection.Vertical({shouldReuse: false}),
+      }),
+    );
+  let selectNewTab =
+    register(
+      "list.selectNewTab",
+      ListSelect({
+        direction: Oni_Core.SplitDirection.NewTab,
```
Comment (bryphe) [inline src/Model/GlobalCommands.re:69]: Looks like this will address #2232 🎉

**Review by bryphe — CHANGES_REQUESTED** [review]
Thanks for taking this on, @CrossR ! Approach looks great to me. Just had a comment about the `Control+V` binding -

File: src/Model/State.re [hunk 4]
```diff
@@ -125,6 +125,21 @@ let defaultKeyBindings =
...
+      bind(
+        ~key="<C-s>",
+        ~command=Commands.List.selectHorizontal.id,
+        ~condition="listFocus" |> WhenExpr.parse,
+      ),
```
Comment (bryphe) [inline src/Model/State.re:137]: This also conflicts with the `save` keybinding, but I think this one is OK - understandable that save wouldn't occur when the quickmenu is up.

**Comment by CrossR** [comment]
Did a quick survey of some common binds for this sort of thing and updated them accordingly, but with Shift-enter as vertical open by default since we use c-v as copy.

That said, obviously if a user wanted to swap they could just do something like:

```json
  { "key": "<C-s>", "command": "list.selectHorizontal", "when": "listFocus" }
```

The only other thing I wanted to check here is if the command names are okay. VSCode has no equivalent, so wasn't sure if they should be prefaced with `oni` or something.

**Comment by bryphe** [comment]
> Did a quick survey of some common binds for this sort of thing and updated them accordingly, but with Shift-enter as vertical open by default since we use c-v as copy.

Awesome! Thanks for looking into this @CrossR . Looks good to me.

> The only other thing I wanted to check here is if the command names are okay. VSCode has no equivalent, so wasn't sure if they should be prefaced with oni or something.

Hmm, good point - VSCode does have a `list.*` set of commands, so prefixing it with `oni.` will make it clear these are Onivim-specific, and keep us safe if VSCode does introduce keybindings with the same name.

**Review by bryphe — APPROVED** [review]

**Comment by CrossR** [comment]
Swapped the names, added some basic docs, just waiting for the CI to be happy then I'll squash/merge.

**Comment by bryphe** [comment]
Sorry about the build issues; the CentOS build failure looks unrelated. I'll override and merge. 

Thanks for the awesome contribution @CrossR !

**Pull request merged**